home/management/commands/updatecourses.py

```diff
@@ -0,0 +1,104 @@
+                for umdio_course in course_data:
+                    course = Course.unfiltered.filter(name=umdio_course['course_id']).first()
+                    if not course:
+                        course = Course(
+                            name=umdio_course['course_id'],
+                            department=umdio_course['dept_id'],
+                            course_number=umdio_course['course_id'][4:],
+                            title=umdio_course['name'],
+                            credits=umdio_course['credits'],
+                            description=umdio_course["description"]
+                        )
+
+                        course.save()
+                        self.total_num_new_courses += 1
+
+                    self._professors(course, s)
+                    print(course)
+
+                kwargs["page"] += 1
+                course_data = requests.get("https://api.umd.io/v1/courses", params=kwargs).json()
+
+        print(f"\n** New Courses Created: {self.total_num_new_courses} **")
+        print(f"** New Professors Created: {self.total_num_new_professors} **")
+
+        runtime = datetime.now() - t_start
+        print(f"Runtime: {round(runtime.seconds / 60, 2)} minutes")
+
+    def _professors(self, course: Course, semester: Semester):
+        kwargs = {"course_id": course.name}
+        umdio_professors = requests.get("https://api.umd.io/v1/professors", params=kwargs).json()
+
+        for umdio_professor in umdio_professors:
+            try:
+                professor = Professor.unfiltered.filter(name=umdio_professor['name']).first()
+            except TypeError:
+                continue
+
+            if umdio_professor['name'] == "Instructor: TBA":
+                continue
+
+            if not professor:
+                # To make our lives easier, attempt to automatically verify the professor
+                # following the same criteria in admin.py
+                split_name = umdio_professor['name'].strip().split()
+                first_name = split_name[0].lower().strip()
+                last_name = split_name[-1].lower().strip()
+                query = Professor.verified.filter(
+                    (
+                        Q(name__istartswith=first_name) &
+                        Q(name__iendswith=last_name)
+                    ) |
+                    Q(slug="_".join(reversed(split_name)).lower())
+                )
+
+                prof_name = " ".join([name.capitalize() for name in umdio_professor['name'].split()])
+                professor = Professor(name=prof_name, type=Professor.Type.PROFESSOR)
+
+                if not query.exists():
+                    professor.slug = "_".join(reversed(split_name)).lower()
+                    professor.status = Professor.Status.VERIFIED
+
+                professor.save()
+                self.total_num_new_professors += 1
+
+            for entry in umdio_professor['taught']:
+                c, s = entry.values()
```

Comment:
wtf is this? you're implicitly relying on the ordering of the fields in the api response, which is NOT guaranteed to be stable. `entry` is a dict with key names that *are* guaranteed to be stable; use them!